### PR TITLE
Fixes #7420: cf-execd is running twice after a reboot or after running rudder-agent twice

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7420

### CODE REVIEW

File: rudder-agent/SOURCES/rudder-agent.init [hunk 1]
```diff
@@ -191,6 +191,17 @@ start_daemons() {
 
 	# It's time to start the daemons
 	for daemon in ${DAEMONS}; do
+    # Test if daemon is already running
+    if [ -s ${CFENGINE_COMMUNITY_PID_FILE[$daemon]} ]
+    then
+      PID=`cat ${CFENGINE_COMMUNITY_PID_FILE[$daemon]}`
+      if ps -p "${PID}" > /dev/null
```
Comment (ncharles) [inline rudder-agent/SOURCES/rudder-agent.init]: why not using PS_COMMAND ?

File: rudder-agent/SOURCES/rudder-agent.init [hunk 2]
```diff
@@ -191,6 +191,17 @@ start_daemons() {
...
+      then
+        message "info" "[INFO] CFEngine Community ${CFENGINE_COMMUNITY_NAME[$daemon]} Is already running"
+        #continue
```
Comment (jooooooon) [inline rudder-agent/SOURCES/rudder-agent.init]: Why is this line commented? We should uncomment it if it's needed or remove it if not.

Comment (peckpeck) [inline rudder-agent/SOURCES/rudder-agent.init]: Sorry, I forgot to uncomment it

File: rudder-agent/SOURCES/rudder-agent.init (hunk 1 above)
Comment (peckpeck) [inline rudder-agent/SOURCES/rudder-agent.init]: PS_COMMAND is for process that have gone wild and have no pid file, it is used to detect and kill them.
Regular process run by the service should have proper pid file, the one we check here.

**Comment by peckpeck** [comment]
PR updated

File: rudder-agent/SOURCES/rudder-agent.init [hunk 3]
```diff
@@ -225,7 +236,7 @@ start_daemons() {
 
 		# Is daemon launched?
 		PID=`cat ${CFENGINE_COMMUNITY_PID_FILE[$daemon]}`
-		if [ ! -e /proc/$PID ]
+		if ! ps -p "${PID}" > /dev/null
```
Comment (jooooooon) [inline rudder-agent/SOURCES/rudder-agent.init]: Why are you changing this here? Is /proc/$PID not available on all systems the reason? Or is it related to this bug?

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Pull request merged**